update to alpaka 0.9.0

psychocoderHPC opened the pull request:
- update git submodule to alpaka 0.9.0 release
```
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="cupla@hzdr.de" git subtree pull --prefix alpaka git@github.com:alpaka-group/alpaka.git 0.9.0 --squash
```
- update documentation
- apply alpaka CMAKE variable name changes
- change alpaka namespace `traits` into `trait`

- update test matrix: test only cuda 11.0 and 11.4 and use container version 2.0
  

Note: container version 2.0 instead of the latest 3.0 is used because using the latest container requires CI refactorings too.

psychocoderHPC committed:
changes required to support alpaka 0.9.0

- update documentation
- apply alpaka CMAKE variable name changes
- change alpaka namespace `traits` into `trait`

sbastrakov reviewed (commented):
I bevieve you didn't add wrappers for new alpaka math functions `arg()` and `conj()`? Or was it a decision to not support those in cupla.

Also, did you check cupla-wrapped math functions work with `alpaka::Complex<>` types?

sbastrakov commented on script/compiler_base.yml:
```diff
@@ -16,10 +16,10 @@
+    ALPAKA_ACCS: "alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE
+                  alpaka_ACC_CPU_B_OMP2_T_SEQ_ENABLE"
+                  # alpaka_ACC_CPU_B_SEQ_T_OMP2_ENABLE=ON
+                  # alpakaACC_CPU_B_SEQ_T_THREADS_ENABLE=ON
```
Should be `alpaka_ACC...`

psychocoderHPC committed:
change CI test combination

Switch to CI container version 2.0. Test CUDA 11.0 and 11.4 only, no
interpediet versions.
Test clang 10 and 11 and gcc 7 to 10.

psychocoderHPC commented:
ready for review

Commit:
Squashed 'alpaka/' changes from 857495c5cd..146b6e6d49

146b6e6d49 Update release date
88e0623c01 Updated documentation for 0.9.0
93f8da8ca0 Prepare release
d651e7fcd4 Extend math unit tests to also check a case of wrapping lambda
7bd6a35e20 Slightly extend accepted margin for math unit tests
50ff0c996b Fix float math tests accidentally using double
46feb03fde Apply a similar fix for alpaka::Complex<> native math implementations
fa69afb66c Add back __host__ __device__ annotations to CUDA/HIP math implementations
ebc1171fea Update CHANGELOG.md
e268708583 Update version number to 0.9.0-rc1
dacd3a2d95 Implement review suggestions
1b6be749e1 Add an example demonstrating use of Complex
beea0bba3b Split math unit tests into multiple translation units
89d31ff666 Extend unit tests for Complex
d41e12d542 Extend math traits for Complex, add new traits for arg() and conj()
9e93f60cce Add alpaka::Complex<T>, mostly a host-device version of std::complex<T>
8d0d179a6a Removed deprecated cudaStreamAddCallback
daa516b5fc Add CudaRt and HipRt template aliases for memory Buffers
48d6a3a0d7 Add CudaRt and HipRt aliases for the Events and Platforms
e1eeb62e26 Set BOOST_LANG_HIP to the HIP version number
069483f400 Change the syntax of saving result of ::omp_get_num_threads()
aa24c90ae0 Fix typos found in code review
eb32e48d12 README_OMP5: Rm cmake flags doc; Add link to readthedocs
d6353ed05d Update OpenMP 5 sphinx usage doc
e3e5cf0269 improve shared pointer handling
3dc811ddcf also run unit tests with zero dimensions
b0f0b333f1 MemFenceOmp5: Rm acq_rel
7559f91c80 Refactor CUDA/HIP queues to reduce code duplication
a07a7dd476 Smarter atomic_ref selection
a1b55c4bce Added CUDA/HIP support for missing types of atomic operations
b3b8016c1c Add SYCL README and fix FPGA CMake options
c8151f928f Do not use hipLaunchKernelGGL
a6cde34505 Merge CudaHipMath back into MathUniformCudaHipBuiltIn
31cfc274f3 fix Sphinx documentation
e2bec6166c fix a warning
57c9f5524e make ALPAKA_ASSERT variadic
60d4472a70 Docs: GTC 2016 Persistent Archive
2a2a824163 Fix typos
3093418d88 Update developer list
a0a378dd7b Update CHANGELOG.md
f5672617a7 CMake: renmae `USE_alpaka_*` to `alpaka_USE_*`
9ec06cb61e Renamed traits namespace to trait
fb44f96141 Lowercase alpaka prefix for CMake variables
b0684629c3 Enforce trivially copyable kernel function objects
343e6de3d6 Fix the host-only compilation for the HIP backend
fad3c7153a alpaka_BUILD_EXAMPLES -> ALPAKA_BUILD_EXAMPLES
f1d9fe2ff9 Refactor getWarpSize to getWarpSizes and return vector
1a8f9b16db CMake version update
48f48be346 Deprecate alpaka::clock()
eff3dcee30 Work around a buggy -Wmissing-variable-declarations warning in clang 5 and 6
fe6fbb5e88 Rewrite the memory fence tests
ced5b3bf9c Update documentation for memory fences
70ccc3ef70 Add a grid-wise memory fence
4189bd1e2d CI version updates
ba7b14486b Enforce trivially copyable kernel arguments
874c210091 retry downloading boost
556077ac19 Merge HIP rows
9348029d74 check that Vec is trivially_copyable
d1a68c78b5 Do not throw in destructors
779ef6f308 Add ROCm 5.0 support
9dcfead9e8 SYCL refactoring and adding native handle support for SYCL queue and event
c2059cad42 Added native handle for EventUniformCUdaHipRt
ff085f23a5 Added native handle specialization for CUDA backend
36e38805c6 add more tests for Vec's ctor
b734f36267 fix a warning with nvcc
517a1da59a Fix SYCL buffer construction
84bf5c63b2 prevent integer underflow in getExtentVecEnd/getOffsetVecEnd
65962de83d allow construction of Vec from types convertible to its value type
60b9ddb365 refactor Vec
d4d73cfb77 Remove incompatible clang + CUDA combinations
afcdeefda5 make alpaka_BUILD_EXAMPLES and alpaka_INSTALL_TEST_HEADER uppercase
4e35ba9b0f update cuda installers
bf9f80e764 test nvcc 11.6 in CI
a8a2d15299 Update URL of the container registry in the documentation
d3e4ab8400 Added draft documentation for writing unit tests
ba6531554e fix calculatePitchesFromExtents for zero dimensional extents
b6734d5e7a rename TPitch to TView when a view is expected
edae13b9c4 delete unneeded move ops on BufCpuImpl/BufOmp5Impl/BufOaccImpl
4b450dc8e4 unit test move ctor/assign of Buf
150f66df40 Fix include path for Intel FPGA extensions
0e835360b2 refactor GetPitchBytesDefault
bb380c2f20 remove several GetPitchBytes specializations
f3c2d8df9b use nested namespace specifier
24e1a289cc make BufCpu class layout not depend on CUDA/HIP
b00dda4441 unify BufUniformCudaHipRt implementation
5fb1416e5a remove pitch from BufCpu
b3d86ac1c9 measure copy times in matMul
6040766f8a add measureRunTimeMs
cb7ac271e4 use delegating ctor for KernelExecutionFixture
7398c05f71 use nested namespace specifiers and avoid unnecessary namespace qualification
99b0ef310d Add experimental SYCL back-end
c6e764c0c8 refactor testing for unsupported types
f060ddf794 refactor common CUDA/HIP code into common header
bbfb856586 use variable template versions of std traits
b1e207f91d remove namespace extent
654cda1c8b use nested namespace specifiers
504a168dce make NormalReal properly copyable
e174566f41 refactor alpaka/rand
cea0c3c7ab Added alias to NativeHandle type and unit test
24425159d4 getNativeHandle functions commited to types
494b5959ce Reverted back to the NativeHandle naming and added common traits for future implementation
80a1ad896d Renamed in getNativeDeviceHandle and added traits for the returned type
8287a27c66 Private constructor for DevUniformCudaHipRt and updated copyrights
bfac39dd6c Added noexcept keyword to getNativeHandle for all the devices
5b428b75f7 Added the getNativeHandle() method for DevCpu (assuming 0 always)
e7eca56090 Renamed iDevice() method for DevOacc
68963a137f Renamed iDevice() method for DevOmp5
fabdb9750a Renamed iDevice() method to getNativeHandle for DevUniformCudaHipRt
99b6fe9c2e Modified device access to DevUniformCudaHipRt
722d552bc7 print all information in GitLab CI job to execute job locally
7b6f4e237e Fix pitch access calculation in randomCells2D example.
a0cfdc14a5 Update BoostPredef.hpp
53b61d0ba4 use nested namespace specifiers
93f25c79e6 Add Visual Studio 17 2022 to CI tests
c9a91352d6 reorganize math headers
9eace7a394 run clang-tidy modernize checks
2d57d41f07 update copyright headers where touched
26cabb4d5a Work around a buggy -Wmissing-variable-declarations warning in clang 5 and 6
c4c4a688a4 Add hasAsyncBufSupport trait, and update tests to use it
4096982a4b TaskKernelOmp5: only print threadElemCount_dev on team 0
0a185e063d ci linux_clang-13_debug_omp5: disable Omp2Blocks ...
dc31aa6dcb AccOmp5: re-format base class list
dc7da1962b BlockSharedMemDynOmp5BuiltIn: Disable MSVC warning
61bdee5a51 ALPAKA_OFFLOAD_USE_BUILTIN_SHARED_MEM: move from enum to structs
3bd3bc51a6 Add test blockSharedMemSharingTest
8b660d6dcb ALPAKA_ASSERT_OFFLOAD: Add do{}while() block for empty case.
8843261983 Add BlockSharedMem{Dyn,St}Omp5BuiltIn
2ea38d6e73 Add a test for the host-only API
6ed9c664ae Support the CUDA/HIP host API without a CUDA/HIP compiler
28747f0599 Fix unreachable code or missing return warnings
a3a299a2ff Use a generator expression for passing options to clang in cuda mode
a231563826 Sphinx Doc: advanced installation documentation
64640119ae Add Accessor-like interface to memory buffers and views
0c098883ac Fix the installation of Boost for the Intel Linux toolkit
106e69a34d Add Boost.Atomic library to CMake
dda46e0bb1 Install the Boost libraries if Boost.Atomic needed
d58629a4bd Add a Boost version check
86e9173e55 Move all CPU accelerators to AtomicCPU
52ac67cd57 Add compile time switch for choosing the atomic implementation
c81f948545 Create a new Atomic by atomic_ref for CPU backends
084ec1e40f fixup 2189d98291: use std::call_once to inite device handles
b4a5fc429e fixup 2189d98291: hold ptr to Impl in DevOacc
1bf4fdb0fe fixup 2189d98291: vector<unique_ptr<DevOaccImpl>>, hold ref in DevOacc
d2e2d28050 fixup 74f9bc69ae: try to fix gitlabci
1322849487 fixup 74f9bc69ae: Add `ALPAKA_CHECK_HEADERS: ON`
07d4a5163c fixup dd8ec93c75: copy: scope buf: use function directly as deleter
24013898b3 fixup 2189d98291 : initialize static devices
2f68dee178 fixup dd8ec93c75: cleanup copy comemnt and scope buf
1f51eca27c fixup: Revert linux_nvhpc-21.9_release_oacc_multicore to one thread
a596fd858a Fix typo in test/unit/kernel/src/KernelWithOmpSchedule.cpp
10c1c99cf3 README_{OMP5,OACC}: Add target host env
9d179ef0fe OAcc CI: Rm fixed CI block size from code:
385353f718 oacc/Copy: Add support for P2P copy
00866b9842 DevOacc: Add for same iDevice return same impl instance every time
b16684680e DevOacc: Fix point of alloc of lock
c63edbaf28 Suppress NVHPC / PGI unreachable code warning
39175f554a Check for internal NVHPC bug number
56539e2378 MapIdx: Add NVHPC loop bug workaround
0d215efa55 Remove alpaka::ignore_unused
826c71a9bd Replace `enable_if` with `if constexpr`
c5b26c88da Update to Catch2 v2.13.8
69469aa0b9 Add integration test for zero-dimensional (scalar) buffers
cc19fac03f Implement OpenAcc memset and copies of zero-dimensional (scalar) buffers
ff085af079 Implement OpenMP memset copies of zero-dimensional (scalar) buffers
a73adb64bd Simplify the dimensionality checks
6f37c69b16 Implement CUDA/HIP memset and copies of zero-dimensional (scalar) buffers
009a9815f6 Implement CPU memset and copies of zero-dimensional (scalar) buffers
fa5feadec2 Add a syntax for printing zero-dimensional Vec
a3a48f792d Add support for zero-dimensional Vec
a956fcbc3a Add support for zero-dimensional (scalar) buffers
506f529b37 Add support for zero-dimensional extents
050d4c2a28 Update .zenodo.json
5d06a5d727 Update CHANGELOG.md
3797b56f05 Update README.md
8e1e4fcbaa Remove MySQL disabling
08d642827d Add changelog for 0.8
f4ca9352bb Use generic container interfaces
544fcbf8c3 Delete meta::apply, replace with std::apply. Delete now unused meta::invoke (can use std::invoke if needed)
487816eeaf Add documentation for setting the HIP GPU architecture via cmake
a1bc2492c5 CI: add ROCm 4.3 and 4.5 tests
609f7652b2 Update tests and examples to use the default memory extent
f7891ad091 Default memory operations to the whole buffer
7f92ec3973 install alpaka as part of CI
247b2d4793 Enabled the serial backend by default for testing and examples
5e008e48cf remove redundant check
9aee4e4f43 fix required boost version in README.md
8bde1c9f6c reformat tables
b1910e864d update Catch2 to development commit e9e4117
a473f4b030 Add clang-13 to CI
66fdd4a3e7 upgrade to at least boost-1.74.0
eba8dff3f1 gitlab CI refactoring
00174b826b Added OpenMP configuration on MacOS
2e3d96ab26 refactor matMul and check pitch
1452063957 CI: gilab paralism
2cfabfd600 CI: speedup boost code download
a4ab9eee9e add cmake option ALPAKA_CHECK_HEADERS
eaf110bac4 CI: OpenACC disable random2D example
79af223cd2 CI: define `ALPAKA_CI` for gitlab
7303a05702 CI: group tests in stages
b6dbd79b03 CI: compile for P5000 GPU
e932466793 CI: run CUDA job on gitlab only
74ea42b32d fix Philox RNG
e46028edba HIP CI node changes
dd1b9af1b6 Remove MySQL disabling
7064fcd05d Add changelog for 0.8
3242b50e9a Remove functions in meta/Metafunctions.hpp, replace with std:: counterparts
b50e217407 Extend a queue's lifetime until async memory operations are complete
53cc0b0ee8 Simplify CPU kernel launch
655b3474cf Silence warnings on non-x86 CPUs
b936b4604b Avoid setting the CUDA/HIP device when not needed
b8b74bb8da Add unit tests for queue-ordered memory buffers
cf886f58f9 Implement queue-ordered memory buffers
8887f06410 Move the allocation logic outside of the BufUniformCudaHipRt class
b562b4644d Move the allocation logic outside of the BufCpuImpl class
aeb50eea77 Update copyright
e016a67394 clang format
4ba61b14bf Isolate tbb::parallel_for
f9b1b7f7b3 CI: fix CUDA 9.2 download path
1ddf48740d remove obsolete workarounds for pre-C++17
a1d6dcc82c switch clang-7/nvcc to clang-8/nvcc
345ef064e4 Use C++17 mandatory copy elision
96c310dd38 remove /cmake/modules from installed directories
c90e7c87df use all cores for building on GitHub runners
673cfa2e97 add nvcc 11.5 to CI
bade5728d4 CI: use ubuntu container and not nvidia container for CUDA
921967661d CI: HIP fix outdated apt keys
9466b3a60d drop support for clang-6/7 with nvcc
e4a76c4dd5 update CUDA installers to ubuntu 20.04
455a9889b2 switch all CI images to ubuntu 20.04
f4f67294ed Delete meta::Void, replace with std::void_t
ac1ccf380d avoid errors with -Wconversion
de18cb8627 update to oneTBB 2021.4.0
4a127c2952 changed linux_nvcc-11.3_gcc-9 CI build from Release to Debug
686d899444 retry a LOT
e1308c8097 add type trait 'remove_restrict'
c0510dfbf5 Add the device type trait specialization for Events
067532b5b7 Added specialization for OpenAcc to disable 8-bit and 16-bit tests
c159ee7fea Replaced function template specialization with function overload for floats and doubles
8cf894d546 Disabled warnings for float comparison with alpaka::math::floatEqualExactNoWarning
0ef767dc1e switch to C++17
aa359876c2 Enable language CXX and update Catch2 message
a90bb9b8e4 fix cheat-sheet RNG section
bed11403e1 Update version number to 0.9.0-develop
cc27006975 CI: git workflows remove HIP tests
164b6b20aa HZDR CI: run HIP tests
edc1aea559 ignore build* directories
147e248ba7 ignore Visual Studio project files (open folder mode)
1b32a518b9 Support non-cached cmake variable ALPAKA_CXX_STANDARD in external projects
01ce3ffb2a Upgrade to Catch2 v2.13.7
e016b775a6 CI: Add NVHPC 21.9 OpenACC
6323250f0b Add nvc++ test exclusions
de9b42dfd1 AtomicOaccExtended: Add NVHPC 21.7 workaround ...
731e80d0ff AtomicTest: Disable 8 and 16-bit tests
2b68034942 Silence NHVPC warnings triggered by catch2
faf28de833 Fix ADL for rsqrt and sincos
b3d199696b Simplify math trait specializations using standard library
06c50db309 fix a typo and add a linebreak in accessor docs
4450298092 add Accessor for views and buffers

git-subtree-dir: alpaka
git-subtree-split: 146b6e6d49083812b3bf40bad5f8d4caf7b031e9

Commit:
Merge commit 'c8e3798b399d2c22fa83ed7c9c3379fa528c4881' into topic-alpaka0.9.0
